packages/dds/tree/src/simple-tree/mapNode.ts

```diff
@@ -244,7 +247,7 @@ export function mapSchema<
 			flexNode: FlexTreeNode,
 		): TreeNodeValid<T2> {
 			if (useMapPrototype) {
-				return new Proxy<Schema>(instance as Schema, handler);
+				return new Proxy<Schema>(instance as Schema, handler as ProxyHandler<Schema>);
```

Comment:
Curious what caused this to be necessary. Is this indicative of something more we need to fix with regards to our typing?

Reply:
MapNode<TreeNodeSchema> used to (and still does for now) allow all possible operations (like setting values to unknown!), and thus a MapNode<TreeNodeSchema> could be used to implement all possible MapNodes with no casts, and horribly unsafe editing APIs. This really should have never worked as the types were giving a really incorrect false sense of safety that was never there. Now the lack of safety is more explicit, and it won't stop building when we fix the implicit unsafety